src/container/intrusive_list.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <cstddef>
+#include <yaclib/container/intrusive_node.hpp>
+
+namespace yaclib::container::intrusive {
+
+template <typename T>
+class List {
+ public:
+  void PushBack(detail::Node<T>* node) noexcept {
+    node->Link(_head._prev, &_head);
+  }
+
+  void PushFront(detail::Node<T>* node) noexcept {
+    node->Link(&_head, _head._next);
+  }
+
+  // Returns nullptr if empty
+  T* PopFront() noexcept {
+    if (IsEmpty()) {
+      return nullptr;
+    }
+    auto* front = _head._next;
+    front->Unlink();
+    return front->AsItem();
+  }
+
+  // Returns nullptr if empty
+  T* PopBack() noexcept {
+    if (IsEmpty()) {
+      return nullptr;
+    }
+    auto* back = _head._prev;
+    back->Unlink();
+    return back->AsItem();
+  }
+
+  // Append (= move, re-link) all nodes from `that` list to the end of this list
+  // Post-condition: that.IsEmpty() == true
+  void Append(List& that) noexcept {
+    if (that.IsEmpty()) {
+      return;
+    }
+
+    auto* that_front = that._head._next;
+    auto* that_back = that._head._prev;
+
+    that_back->_next = &_head;
+    that_front->_prev = _head._prev;
+
+    auto* back = _head._prev;
+
+    _head._prev = that_back;
+    back->_next = that_front;
+
+    that._head._next = that._head._prev = &that._head;
+  }
+
+  bool IsEmpty() const noexcept {
+    return _head._next == &_head;
+  }
+
+  List() {
+    InitEmpty();
+  }
+
+  List(List&& that) noexcept {
```

Comment:
- [ ] Same problem

Or move ctor + assign
Or methods

NIT
I prefer the option with methods